test/unit-test/dawn/SIR/TestASTVisitor.cpp

```diff
@@ -0,0 +1,245 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
```

Comment:
should we have all the helper-classes up top and all the tests afterwards?

Reply:
sorry I didnt understand. What do you mean?

Reply:
The question of what better practice is:
```
class used_in_test_1{
 ...
}
class used_in_test_2{
  ...
}

TEST_F(tests, used_in_test_1){
 ...
}
TEST_F(tests, used_in_test_2){
 ... 
}
```
or
```class used_in_test_1{
...
}
TEST_F(tests, used_in_test_1){
...
}
class used_in_test_2{
...
}

TEST_F(tests, used_in_test_2){
...
}
```
Probably v1 is easier if you want to check what tests are doing whereas v2 has higher readability if you need to debug something.